pkg/operator/operator.go

```diff
@@ -21,28 +23,105 @@ const instanceSelector = "app.kubernetes.io/managed-by=observability-operator"
 
 const ObservabilityOperatorName = "observability-operator"
 
+// The default values we use. Prometheus and Alertmanager are handled by
+// prometheus-operator. For thanos we use the default version from
+// prometheus-operator.
+var DefaultImages = map[string]string{
+	"prometheus":   "",
+	"alertmanager": "",
+	"thanos":       "quay.io/thanos/thanos:" + obopo.DefaultThanosVersion,
+}
+
+func ImagesUsed() []string {
+	i := 0
+	imgs := make([]string, len(DefaultImages))
+	for k := range DefaultImages {
+		imgs[i] = k
+		i++
+	}
+	return imgs
+}
+
 // Operator embedds manager and exposes only the minimal set of functions
 type Operator struct {
 	manager manager.Manager
 }
 
-func New(metricsAddr, healthProbeAddr string) (*Operator, error) {
+type OperatorConfiguration struct {
+	MetricsAddr     string
+	HealthProbeAddr string
+	Prometheus      stackctrl.PrometheusConfiguration
+	Alertmanager    stackctrl.AlertmanagerConfiguration
+	ThanosSidecar   stackctrl.ThanosConfiguration
+	ThanosQuerier   tqctrl.ThanosConfiguration
+}
+
+func WithPrometheusImage(image string) func(*OperatorConfiguration) {
+	return func(oc *OperatorConfiguration) {
+		oc.Prometheus = stackctrl.PrometheusConfiguration{Image: image}
```

Comment:
in case we extend the Prometheus config in the future
```suggestion
		oc.Prometheus.Image = image
```

Reply:
Should we go and adapt this same approach for the other components as well? Not sure how likable for us to extend their configuration, though